### PR TITLE
Ability to nest forms, new defaultLocked prop

### PR DESCRIPTION
**Pull request by suprMax**

### CODE REVIEW

File: preview/containers/locked/index.es [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            <FormGroup>
+              <Input
+                type="text" label="Pattern validation - 'abc'" name="pattern_abc"
+                error="Value incorrect!" pattern="abc" required
+              />
+            </FormGroup>
+            <FormGroup>
+              <Input
+                type="text" label="Pattern validation - '\d{5}'" name="pattern_numeric"
+                error="Value incorrect!" pattern="\d{5}" required
```
Comment (atomiomi) [inline preview/containers/locked/index.es:78]: what if there is an optional field that a user won't touch?

File: src/form/index.es [hunk 2]
```diff
@@ -165,21 +189,42 @@ class Form extends PureComponent {
...
+      };
+
+      if (isNativeForm) {
+        buttonProps.type = 'submit';
+        buttonProps.disabled = isDisabled;
+      } else {
+        buttonProps.onClick = this.submit;
+      }
+
+      button = <ButtonComponent {...buttonProps}>{buttonText}</ButtonComponent>;
```
Comment (atomiomi) [inline src/form/index.es:227]: move to separate render function? or maybe component?

File: preview/containers/locked/index.es (hunk 1 above)
Comment (suprMax) [reply preview/containers/locked/index.es:78]: good point, will think about it, might have to have it as a known issue for this option

File: src/form/index.es (hunk 2 above)
Comment (peterhass) [reply src/form/index.es:227]: +1 for the render function

File: src/form/index.es [hunk 3]
```diff
@@ -219,7 +269,8 @@ Form.propTypes = {
   onInvalidSubmit: PropTypes.func,
   onSubmit: PropTypes.func,
 
-  locked: PropTypes.bool,
+  locked: PropTypes.bool.isRequired,
+  defaultLocked: PropTypes.bool.isRequired,
```
Comment (peterhass) [inline src/form/index.es:273]: `lockedByDefault` sounds nicer imo

Comment (suprMax) [reply src/form/index.es:269]: we have a common pattern of prefixing default value for a prop with `default`. I see how yours sounds clearer but for consistency I think it's best to use `defaultSmth`

File: preview/containers/locked/index.es (hunk 1 above)
Comment (suprMax) [reply preview/containers/locked/index.es:84]: solved... i think

File: src/form/button.es [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+  return <Component {...props}>{text}</Component>;
+};
+
+Button.defaultProps = {
+  as: 'button',
+  disabled: false,
+};
+
+Button.propTypes = {
```
Comment (atomiomi) [inline src/form/button.es:28]: bettername like Submit?

Comment (suprMax) [reply src/form/button.es:28]: internal subcomponent, imo `button` is pretty good

**Comment by suprMax** [comment]
Bug: errors set by server don't clear locked state

**Comment by suprMax** [comment]
The only way to accomplish this task was to revalidate all of the inputs all the time (every update, every time error is cleared etc.).

This PR is parked for now, if PMs come with a requirement for locking of forms by default once again – this can be merged.

**Pull request closed without merging**